reframe/core/runtime.py

```diff
@@ -204,6 +213,17 @@ def mode(self, name):
         except KeyError:
             raise ConfigError('unknown execution mode: %s' % name) from None
 
+    def next_run(self):
+        self._current_run += 1
+
+    @property
+    def current_run(self):
```

Comment:
What about calling this property `runid`? I think it is better from the user's point of view. It will also help explaining it in the documentation.

Reply:
Having looked the whole PR, I think we can do the following:

```python
class RuntimeContext:
    def __init__(...):
        self._runid = 0

    @property
    def runid(self):
        """Fully documented, since it is meant to be used also be regression tests"""
        return self._runid

    @property
    def run_count(self):
        # Not publicly documented
        return self._runid
```

Then, in places where conceptually you need the count, you call `run_count`, whereas in places where the `id` is more meaningful you call `runid`.